### ISSUE TITLE
Google coral dev board - object detection using Yolo

### ISSUE THREAD

**Issue by MECHAAMER** [google-coral/edgetpu#868]
Hello everyone,

I'm working on training a Yolo model for object detection and plan to use a Google Coral Dev Board for inference. As the Coral documentation recommends, the model should be in the TFLite format with 8-bit quantization for optimal performance.

Thanks to Ultralytics, exporting the model to the required format is straightforward:

```
from ultralytics import YOLO
model = YOLO("pre_trained_model.pt")

# Export the model to TFLite Edge TPU format
model.export(format="edgetpu")
```

In the output, I see:

    Number of operations that will run on Edge TPU: 425
    Number of operations that will run on CPU: 24

My question is: Can I do anything to make all operations run on the TPU for faster processing?

Additionally, are there any other recommended models that might offer better accuracy and lower latency on a Google Coral board?

Thanks all.

**Comment by ajbolt69** [comment]
Try using mobilenet v2 ssd

**Comment by MECHAAMER** [comment]
is it better than Yolov8n?

**Comment by ajbolt69** [comment]
depends on your specific use case, did you try running the yolov8n on the
devboard already? what performance stats are you getting? inference times
etc.

**Comment by MECHAAMER** [comment]
Thanks for reply, I am outside office now and don't have access to the test data, however the inference was slow, need 0.14 sec per frame. I think the inference will be faster if I can run all the ops in the TPU

**Comment by MECHAAMER** [comment]
Yes I tired to run the Yolov8n, the accuracy almost good, but the latency high, 5.58 frames/second only, do you think mobilenet v2 ssd accuracy will be faster and more accurate?

**Comment by ajbolt69** [comment]
I don’t have the stats to comment on accuracy , but inference time will be
much faster , it’s easily running at 30fps

**Comment by MECHAAMER** [comment]
I read this sentence on the coral official document: 

"If part of your model executes on the CPU, you should expect a significantly degraded inference speed compared to a model that executes entirely on the Edge TPU. We cannot predict how much slower your model will perform in this situation, so you should experiment with different architectures and strive to create a model that is 100% compatible with the Edge TPU. That is, your compiled model should contain only the Edge TPU custom operation."

you suggest use mobilenet v2 ssd, do you have another recommendation for a models that can work 100% on the TPU (object detection use)?

**Comment by ajbolt69** [comment]
do you have any compatibility issues running the mobilenet v2 ssd ? i
suggest you try it out for yourself and decide if you like it or not, in my
opinion it runs smoothly , optimised for the tpu , thats the reason
https://github.com/google-coral/examples-camera

**Comment by MECHAAMER** [comment]
thank you for your reply and time, I will try it and comeback with results.

**Comment by MECHAAMER** [comment]
do you recommenced and training parameters for the Mobilenet v2 SSD? is there any link can help me ?

the link you attached recently explain how to use a pre-trained model, I want to train the model with my dataset.

**Comment by ajbolt69** [comment]
I’m not very sure about that , I have just used the pretrained model as
well , I’ll be working on custom training for the next 2 weeks

**Comment by MECHAAMER** [comment]
ok thank you, do you recommend use EfficientDet or mobilenet for object detection?

**Comment by ajbolt69** [comment]
Hey sorry for the late reply, i would recommend the mobilenet since i have used it it uses the edge tpu well and runs at 30fps easily , 20ms inference time per frame , currently working on deploying a custom yolo model to the devboard and Rpi5 , would you like to connect separately? arinjay1402@gmail.com , thats my email

**Comment by Curid** [comment]
mobilenet v2 has a [latency of 7.6ms](https://codeberg.org/Curid/TF-CCTV) on the edgetpu, which would mean the maximum theoretical framerate is 130fps

**Comment by adamp87** [comment]
Hi, if i remember correctly YOLOv8n with a smaller input image size can have all operations on the TPU. e.g. `model.export(format="edgetpu", imgsz=192)`. If you visualize your model with Netron, you can see the Transpose operation to break the TPU execution, probably due to the large array size. But with same input size, I wouldnt expect such a large speedup, these are only the last operations and the majority of the expensive Conv2D etc operations are done on the TPU. Ofc, with a smaller input size you will have less operations, so expect speedup but a drop in mAP.

I personally use the SSDLite MobileDet from Coral AI as this model has a nice accuracy and speed.
